test/test_utils/pytest_cache.py

```diff
@@ -133,6 +138,32 @@ def upload_pytest_cache_from_local_to_s3(self,
         s3_file_path = os.path.join(s3_file_dir, "lastfailed")
         self.__upload_cache_to_s3(local_file_path, s3_file_path)
 
+    def convert_cache_json_and_upload_to_s3(self,
+                                            cache_json,
+                                            commit_id,
+                                            framework,
+                                            version,
+                                            build_context,
+                                            test_type):
+        """
+        Copy pytest cache from json and send to directory in s3. 
+        :param cache_json - json object with pytest cache
+                Following parameters are required to create a path to cache file in s3:
+        :param commit_id
+        :param framework
+        :param version
+        :param build_context
+        :param test_type
+        """
+        if len(cache_json) == 0:
+            LOGGER.info("No cache was generated. Skip uploading.")
+            return
+        s3_file_dir = self.__make_s3_path(commit_id, framework, version, build_context, test_type)
+        s3_file_path = os.path.join(s3_file_dir, "lastfailed")
+        with open("tmp_file_for_cache_json", "w") as f:
+            f.write(json.dumps({**cache_json}))
```

Comment:
Please use `json.dump(<object>, f)` rather than using `f.write(json.dumps(<object>))`.

Also, please use `dict(cache_json)` to convert it from `DictProxy` to `dict`, rather than `{**cache_json}`, which is a roundabout and slightly confusing way of doing the same.

```suggestion
            json.dump(dict(cache_json), f)
```